source/windows/iocp/socket.c

```diff
@@ -805,29 +811,28 @@ static inline int s_tcp_connect(
     uint64_t time_to_run = 0;
     /* if the connect succedded immediately, let the timeout task still run, but it can run immediately. This is cleaner
        because it can just deallocate the memory we just allocated. */
-    aws_event_loop_current_clock_time(socket->event_loop, &time_to_run);
+    aws_event_loop_current_clock_time(connect_loop, &time_to_run);
 
     /* with IO completion ports, the overlapped callback triggers even if the operation succedded immediately,
        so we can just act like it's pending and the code path is the same.*/
     if (!connect_res) {
         int error_code = WSAGetLastError();
         if (error_code != ERROR_IO_PENDING) {
-            socket_impl->read_io_data->in_use = false;
-            aws_mem_release(socket->allocator, connect_args);
             socket_impl->connect_args = NULL;
-            socket->state = ERRORED;
+            socket_impl->read_io_data->in_use = false;
```

Comment:
if the `socket` might be killed from callbacks above, then won't the `socket_impl` also be deleted?

Reply:
yeah, this was done when we believed that the completion port didn't get invoked in an error condition. Since we now know it does, most of this branch can just be deleted.